tests/test_lambdas/test_client.py

```diff
@@ -220,3 +244,108 @@ def test__invoke__returns_accepted_result_when_invoked_asynchronously(
 
     assert result == {"result": LambdaResult.ACCEPTED}
     assert caplog.messages == []
+
+
+def test__request__raises_error_when_response_could_not_be_read_correctly(
+    lambda_client: MagicMock, caplog: LogCaptureFixture
+) -> None:
+    response = {"StatusCode": 200, "Payload": "invalid-payload"}
+    lambda_client.invoke.return_value = response
+
+    with pytest.raises(AttributeError):  # type of error does not matter here
+        LambdaClient.request(
+            "test-func",
+            "GET",
+            "/home",
+        )
+
+    logged_record = caplog.records[0]
+    assert logged_record.message == "Invalid response received from test-func Lambda (path: /home)"
+
+
+def test__request__returns_response_when_requesting_with_no_body(
+    lambda_client: MagicMock,
+) -> None:
+    payload = rest_response_factory(body={})
+
+    lambda_client.invoke.return_value = payload
+
+    result = LambdaClient.request("test-function", "/home", "GET")
+
+    assert result.to_dict() == {
+        "body": "{}",
+        "headers": {},
+        "isBase64Encoded": False,
+        "statusCode": 200,
+    }
+    lambda_client.invoke.assert_called_once_with(
+        FunctionName="test-function",
+        Payload=ANY,  # Any idea how to generate predefined requestId
+        InvocationType="RequestResponse",
+    )
+
+
+def test__request__returns_response_when_requesting_with_body(
+    lambda_client: MagicMock,
+) -> None:
+    payload = rest_response_factory(body={"adult": "child"})
```

Comment:
These tests are incorrect as the only thing that you change is the payload that is returned from the mocked method 🤷